Secrets loaded with IO.read need newline characters stripped

taharah opened the pull request:
Encrypted data bag secrets can contain newline characters that need to be removed in order to successfully load the data bag item.

chef-delivery commented:
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

jamescott commented:
@delivery approve

chef-delivery commented:
Change: f1bb245c-241d-43cb-b558-5f2f3564a802 approved by: @jamescott

chef-delivery commented:
This PR has passed 'Acceptance' and is ready to be delivered!
  Use: '@delivery deliver' when validated in acceptance.

taharah committed:
Secrets loaded with IO.read need newline characters stripped

Encrypted data bag secrets can contain newline characters that need to be removed in order to successfully load the data bag item.

Commit:
adjustments / dtags replicate for PR728

brewn commented:
Thanks @taharah, I've PR'd some changes here:

https://github.com/taharah/chef-web-docs/pull/1

brewn commented:
Err, looks like this may already be covered here:

https://github.com/chef/chef-web-docs/pull/729

taharah commented:
Using the method `Chef::EncryptedDataBagItem.load_secret` clashes with the Foodcritic rule [FC086](http://www.foodcritic.io/#FC086). The `load_secret` method itself performs a strip when reading the file ([encrypted_data_bag_item.rb#L147](https://github.com/chef/chef/blob/master/lib/chef/encrypted_data_bag_item.rb#L147)), so this should achieve the same functionality while not violating FC086.

taharah committed:
Merge pull request #1 from brewn/728-check

adjustments / dtags replicate for PR728
